File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_026)_Maximum_Consecutive_ONEs_in_binary_Array.cpp
#include<iostream>
using namespace std;

int maxConsecutiveOnes(bool arr[], int n)
{
    int res = 0, curr = 0;
    for(int i=0; i<n; i++)
    {
        if(arr[i] == 0)
            curr = 0;
        else
        {
            curr++;
            res = max(res, curr);
        }
    }
    return res;
}

int main()
{
    bool arr[] = {0, 1, 1, 0, 1, 1, 1};
    int n=7;
    cout<<"Maximum 1's in binary Array = "<<maxConsecutiveOnes(arr, n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_026)_Maximum_Consecutive_ONEs_in_binary_Array.dart
import 'dart:math';

int maxConsecutiveOnes(arr) {
  var res = 0;
  var count = 0;
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] == 0) {
      count = 0;
    } else {
      count++;
      res = max(res, count);
    }
  }
  return max(res, count);
}

void main() {
  var arr = [0, 1, 1, 0, 1, 1, 1];
  print('Max consecutive ones in binary array is ${maxConsecutiveOnes(arr)}.');
}
